src/Stratis.Bitcoin/Base/BackgroundWork/BackgroundWorkProvider.InnerClasses.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Stratis.Bitcoin.Signals;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Base.BackgroundWork
+{
+    public partial class BackgroundWorkProvider : IBackgroundWorkProvider
+    {
+        /// <summary>
+        /// Interface used to allow BackgroundWorkProvider to set private properties
+        /// </summary>
+        private interface IAsyncTaskInfoSetter
+        {
+            TaskStatus Status { set; }
+
+            Exception Exception { set; }
+        }
+
+        /// <summary>
+        /// Class that holds the status of running or faulted async delegate created by the BackgroundWorkProvider
+        /// </summary>
+        /// <seealso cref="Stratis.Bitcoin.Base.BackgroundWork.BackgroundWorkProvider.IAsyncTaskInfoSetter" />
+        internal class AsyncTaskInfo : IAsyncTaskInfoSetter
+        {
+            public string FriendlyName { get; }
```

Comment:
Should we not have a `FaultBehaviour` flag? 
- shutdown node
- shutdown component
- do nothng

Reply:
that part is still to be implemented, I injected both INodeLifetime and ISignals and we need to decide what to do now
Anyway I don't see why adding "shutdownnode" in that flag since when you set it...you are shutting down it so you don't make any use of that information